src/Sarif/Baseline/V2/WhatComponent.cs

```diff
@@ -42,19 +42,30 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            int hashCode = 0;
+            int hashCode = 17;
 
-            hashCode *= 13;
-            hashCode = this.Category?.GetHashCode() ?? 0;
```

Comment:
>0 [](start = 55, length = 1)

Bug: We were resetting `hashCode` to 0 each time we found a `null` property value.